aten/src/ATen/native/cpu/FusedSGDKernel.cpp

```diff
@@ -142,20 +141,20 @@ typename std::enable_if<
     }
     if (maximize) grad_vec = grad_vec * Vec(scalar_t(-1.0));
     if (weight_decay != 0.0){
-      grad_vec += param_vec * Vec(scalar_t(weight_decay));
+      grad_vec = vec::fmadd(param_vec, Vec(scalar_t(weight_decay)), grad_vec);
     }
     if (momentum != 0.0) {
       Vec momentum_vec;
       if (is_first_step) {
         momentum_vec = grad_vec;
       } else {
         momentum_vec =
-            Vec::loadu(momentum_buf_ptr + d) * Vec(scalar_t(momentum)) +
-            grad_vec * Vec(scalar_t(1 - dampening));
+            Vec::loadu(momentum_buf_ptr + d) * Vec(scalar_t(momentum));
+        momentum_vec = vec::fmadd(Vec(scalar_t(1 - dampening)), grad_vec, momentum_vec);
```

Comment:
for my understanding: what’s the difference between vec::fmadd vs += and a *? I’m guessing it’s for performance (this is a fused kernel) but does it also do more vectorization?

Reply:
I think there might be some rounding differences between `fma` and  `+ and *`.